kotlinx-coroutines-core/jvm/src/internal/Concurrent.kt

```diff
@@ -36,3 +36,5 @@ internal fun removeFutureOnCancel(executor: Executor): Boolean {
         return false // failed to setRemoveOnCancelPolicy, assume it does not removes future on cancel
     }
 }
+
+internal actual val multithreadingSupported: Boolean = true
```

Comment:
I'd suggest doing so when we are going to evolve the code; it seems a bit premature for now

Reply:
Having searched my feelings for why this bugs me, I think I have a more compelling explanation.

This variable is present in the non-test common code, even though it wasn't used anywhere other than on the Native. This, like not making a variable `private` when it can be, is, I think, an anti-pattern: making the scope in which something is present large is easy, reducing the scope is not.

What happened just now, in this PR, is an example of just that: this variable only concerns the behavior on Native, but, due to it being available everywhere, it is used in the common code. If we later make Native multithreaded by default, we may not notice that the variable outlived its usefulness and can be removed everywhere but in tests (and possibly even there).

> it seems a bit premature for now

I'd argue the opposite: it's premature to have a broad scope of availability before it's actually needed.

Reply:
These are good points!

I've made the following: made `runMtTest` expect function only in `concurrent` source-set, made `multithreadingSupported` native-only declaration and moved the only use of `multithreadingSupported` in `common` to Native source-set